secretstores/azure/keyvault/authutils.go

```diff
@@ -37,7 +37,7 @@ func (s EnvironmentSettings) GetClientCert() (CertConfig, error) {
 	clientID := s.Values[componentSPNClientID]
 	tenantID := s.Values[componentSPNTenantID]
 
-	if certFilePath == "" && certBytes == nil {
```

Comment:
can `s.Values[componentSPNCertificate]` be nil?  then certBytes is always zero byte array?

Reply:
`Values` is `map[string]string` which does not allows for null and returns an empty string when the given key is not present.

Guess a better check would be:

``` 
        certBytes, bytesPresent := s.Values[componentSPNCertificate]
	certPassword := s.Values[componentSPNCertificatePassword]
	clientID := s.Values[componentSPNClientID]
	tenantID := s.Values[componentSPNTenantID]

	if certFilePath == "" && !bytesPresent {
		return CertConfig{}, fmt.Errorf("missing client secret")
	}
```

thoughts?

Reply:
looks good to me :)

Reply:
Fixed in f3e67d1